skins/cat17/address-form/src/store.ts

```diff
@@ -16,78 +16,85 @@ const ADDRESS_FIELD_VALIDATOR_NAMES: Array<string> = [
-		},
-		[FINISH_ADDRESS_VALIDATION](state, payload) {
-            if (payload.status === 'ERR') {
-                ADDRESS_FIELD_VALIDATOR_NAMES.forEach(name => {
-                	if ( payload.messages[name] ) {
-						state.form[name] = Validity.INVALID;
-					}
-                });
-            }
-			state.isValidating = false;
+	[MARK_EMPTY_FIELD_INVALID](state, payload) {
+		Object.keys(payload).forEach((field: string) => {
+			const fieldName = payload[field];
+			if (!fieldName.optionalField && state.form[fieldName.name] === Validity.INCOMPLETE) {
+				state.form[fieldName.name] = Validity.INVALID;
+				}
+			});
+	},
+	[BEGIN_ADDRESS_VALIDATION](state) {
+		state.isValidating = true;
+	},
+	[FINISH_ADDRESS_VALIDATION](state, payload) {
+		if (payload.status === 'ERR') {
+			ADDRESS_FIELD_VALIDATOR_NAMES.forEach(name => {
+				if ( payload.messages[name] ) {
+					state.form[name] = Validity.INVALID;
+				}
+			});
 		}
+		state.isValidating = false;
+	}
+}
+
+export const getters: GetterTree<AddressState, any> = {
+	invalidFields: (state: AddressState) :Array<string> => {
+		return Object.keys(state.form).filter(field => state.form[field] === Validity.INVALID)
 	},
-	getters: {
-		invalidFields: (state) :Array<string> => {
-			return Object.keys(state.form).filter(field => state.form[field] === Validity.INVALID)
-		},
-		allFieldsAreValid: (state, getters) :boolean => {
-			return getters.invalidFields.length === 0;
-		},
+	allFieldsAreValid: (state: AddressState, getters: GetterTree<AddressState, any> ) :boolean => {
+		return getters.invalidFields.length === 0;
+	}
+}
+
+export const actions: ActionTree<AddressState, any> = {
+	validateInput({ commit }, field: InputField) {
+		commit('VALIDATE_INPUT', field);
 	},
-	actions: {
-		validateInput({ commit }, field: InputField) {
-			commit('VALIDATE_INPUT', field);
-		},
-		storeAddressFields({ commit, getters }, payload: Payload) {
-			commit('MARK_EMPTY_FIELD_INVALID', payload.formData);
-			if ( getters.allFieldsAreValid ) {
-                commit('BEGIN_ADDRESS_VALIDATION');
-                const postData = Helper.formatPostData(payload.formData);
-                return payload.transport.postData(payload.validateAddressURL, postData)
-					.then( (validationResult: ValidationResult) => {
-						commit('FINISH_ADDRESS_VALIDATION', validationResult);
-					});
-			}
-			return Promise.resolve();
+	storeAddressFields({ commit, getters }, payload: Payload) {
+		commit('MARK_EMPTY_FIELD_INVALID', payload.formData);
+		if ( getters.allFieldsAreValid ) {
+			commit('BEGIN_ADDRESS_VALIDATION');
+			const postData = Helper.formatPostData(payload.formData);
+			return payload.transport.postData(payload.validateAddressURL, postData)
+				.then( (validationResult: ValidationResult) => {
+					commit('FINISH_ADDRESS_VALIDATION', validationResult);
+				});
 		}
+		return Promise.resolve();
 	}
 }
 
+const store: StoreOptions<AddressState> = {
+	state,
+	mutations,
+	getters,
+	actions
+}
```

Comment:
I like the split into individual constants, that gets rid of some of the nesting and can make things more visible and navigatable in the IDE.